python/lsst/ts/GenericCamera/driver/canoncamera.py

```diff
@@ -0,0 +1,145 @@
+            The name of the configuration file to load."""
+        self.id = config.id
+        self.binValue = config.binValue
+        self.normalImageType = config.ImageType
+        self.currentImageType = self.normalImageType
+
+        # Initialize the camera. If not camera is detected then an Exception
+        # will be raised.
+        self.camera = gp.Camera()
+        self.camera.init()
+
+    def getMakeAndModel(self):
+        """Get the make and model of the camera.
+
+        Returns
+        -------
+        str
+            The make and model of the camera."""
+        abilities = self.camera.get_abilities()
+        return abilities.model
+
+    def getROI(self):
+        """Gets the region of interest.
+
+        Returns
+        -------
+        int
+            The top most pixel of the region.
+        int
+            The left most pixel of the region.
+        int
+            The width of the region in pixels.
+        int
+            The height of the region in pixels."""
+        return 0, 0, 6720, 4480
+
+    async def startTakeImage(self, expTime, shutter, science, guide, wfs):
+        """Start taking an image or a set of images.
+
+        Parameters
+        ----------
+        expTime : float
+            The exposure time in seconds.
+        shutter : bool
+            Should the shutter be opened?
+        science : bool
+            Should the science/main sensor be used?
+        guide : bool
+            Should guider sensor be used?
+        wfs : bool
+            Should wave front sensor be used?
+        """
+        # In order to be able to update the cameraq config via GPhoto, the
+        # config object needs to be obtained ...
+        cfg = self.camera.get_config()
+        # ... the config parameters need to be adjusted ...
+        iso = cfg.get_child_by_name("iso")
+        iso.set_value("100")
+        shutterspeed = cfg.get_child_by_name("shutterspeed")
+        shutterspeed.set_value(str(expTime))
+        # ... and the config needs to be written baqck to the camera
+        self.camera.set_config(cfg, None)
+        await super().startTakeImage(expTime, shutter, science, guide, wfs)
+
+    async def startIntegration(self):
+        """Start integrating.
+        """
+        # This starts the exposure while obtaining the path to the file on the
+        # camera. The exposure will finish by itself because the camera
+        # controls that.
+        self.file_path = self.camera.capture(gp.GP_CAPTURE_IMAGE)
+        await super().startIntegration()
+
+    async def endReadout(self):
+        """End reading out the image.
+        """
+        # The image can be obtained by reading out the file_path variable in
+        # which the path to the file on the camera was stored.
+        target = os.path.join("/tmp", self.file_path.name)
+        camera_file = self.camera.file_get(self.file_path.folder, self.file_path.name, gp.GP_FILE_TYPE_NORMAL)
+        camera_file.save(target)
```

Comment:
Do your really need to store it in a file to be able to read it? what kind of object is `camera_file`?